atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -299,12 +297,16 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         KeyValueService keyValueService = initializeCloseable(() -> {
             KeyValueService kvs = atlasFactory.getKeyValueService();
             kvs = ProfilingKeyValueService.create(kvs);
-            kvs = SweepStatsKeyValueService.create(kvs,
-                    new TimelockTimestampServiceAdapter(lockAndTimestampServices.timelock()),
-                    JavaSuppliers.compose(SweepConfig::writeThreshold, sweepConfig),
-                    JavaSuppliers.compose(SweepConfig::writeSizeThreshold, sweepConfig)
 
-            );
+            // If we are writing to the sweep queue, then we do not need to use SweepStatsKVS to record modifications.
+            if (!config.targetedSweep().enableSweepQueueWrites()) {
```

Comment:
Do we think this is sufficient? This means that writes are being persisted to the queue, but targeted sweep might not be enabled - we could check both. I can't think of a good reason to stay in that mode, unless we're addressing some critical bug that occurs when targeted sweep processes the queue. Even in that case, this would be a temporary state.

Reply:
You're right that writing to the sweep queue but not processing it should be a temporary state - just like the state where we are writing sweep stats to `sweep.priority` (i.e. `enableSweepQueueWrites` is false), but sweep is currently disabled.

The large internal product enables and disables sweep on a daily basis due to banned hours config, and so might do the same with targeted sweep. It would seem silly to write sweep stats for the entire lifecycle of a KVS, just because it was started at a time when targeted sweep was turned off.

Reply:
In fact, enabling sweep.priority writes in this case would actually be harmful, because the background sweeper would end up doing unnecessary work.